gherkin/GHERKIN_MARKDOWN.md

```diff
@@ -0,0 +1,38 @@
+# Gherkin Markdown
+
+Gherkin Markdown is not a Markdown dialect per se, but rather a set of rules for
+extracting scenarios from Markdown documents.
+
+The rules are as follows:
+
+- `#` - Feature
+- `##` - Scenario, Background or Rule, depending on context
+- `###` - Examples, Scenario or Background, depending on context
+- `####` - Examples
+- `* {Keyword}` - Given, When, Then, And and But
+- `|` - Tables (DataTable and ExamplesTable)
+- `\`\`\`` - DocString
+- `>` - prefix for @tags
```

Comment:
The quotes may be not very appropriate for tags, they maybe useful for descriptions of feature or scenarios.

Please consider to allow tags in html comments, especially for features, e.g.

```
<!-- @tag @wip -->
# Feature

<!-- @foo -->
## Scenario
```

Reply:
Using `>` for descriptions of scenarios and features wouldn't be necessary. You'd just use normal paragraphs for that:

```markdown
## This is a scenatio
This is the description
over a few lines...

* a step
* another step
```

Are there other reasons we want to consider something else than `>` to prefix tags?

If we put tags inside HTML comments, they won't be rendered once the Markdown is converted to HTML, and I think most users would expect them to be rendered.

Reply:
Sorry, I maybe not very clear with my bad English. I did not mean `>` as marker for feature description, I mean `>` is a quote in markdown and I imagine I would want to add some quotes in the document which describes the a feature.

> If we put tags inside HTML comments, they won't be rendered once the Markdown is converted to HTML, and I think most users would expect them to be rendered.

Yeah, this is what I also came to after thinking more about my proposal. 

In general I think the `>` is fine (if parser will look tags which starts with `@` in the quotes).

The only thing that unclear to me, is the parser going to look for tags before `# Feature` or after? Because I think the quotes before level 1 header is not how we usually write the documents.

Reply:
Yeah, I think it would look better if the tags came after the heading. However, that changes the grammar a bit and might complicate things.

This could work:

```markdown
# @foo @bar
# Feature: Hello
According to research:
> Water boils at 200C

## @zap
## Scenario: Hello
```

It admittedly looks a bit weird in Markdown and the default rendering, but we could style it so that `# @tag` headers are rendered more like small tags above the real header.